macros/src/lib.rs

```diff
@@ -28,13 +28,11 @@ pub fn reflike(input: TokenStream) -> TokenStream {
     let lit = parse_macro_input!(input as LitStr);
 
     match RefLike::try_from(lit.value()) {
-        Ok(safe) => {
-            let safe_str = safe.as_str();
+        Ok(_) => {
             let expand = quote! {
-                unsafe {
-                    ::std::mem::transmute::<::std::path::PathBuf, ::radicle_git_ext::RefLike>(
-                        ::std::convert::From::from(#safe_str)
-                    )
+                {
+                    use ::std::convert::TryFrom;
```

Comment:
What's the reason you removed the `transmute`? We have already computed a valid `RefLike`, so don't have to do it again at runtime.

Reply:
The transmute was converting from a PathBuf, so I thought we could remove it, as it still has a From. Maybe it could could straight into the RefLike but I didn't think about it at first, only about removing the PathBuf from the code

Reply:
Yeah, it can just transmute the `safe_str` into a `RefLike`. Alternatively, we can also add a constructor marked `unsafe` on `RefLike` itself, which doesn't validate the `String` argument.